kong/plugins/opentelemetry/utils.lua

```
@@ -48,8 +57,118 @@ local function get_headers(conf_headers)
+end
+
+local function read_resource_attributes(resource_attributes)
+  if not resource_attributes then
+    return EMPTY
+  end
+  local __meta_environment = {
+    __index = function(self, key)
+      local lazy_loaders = {
+        headers = function(self)
+          return get_request_headers() or EMPTY
+        end,
+        query_params = function(self)
+          return get_uri_args() or EMPTY
+        end,
+        uri_captures = function(self)
+          return (ngx.ctx.router_matches or EMPTY).uri_captures or EMPTY
+        end,
+        shared = function(self)
+          return ((kong or EMPTY).ctx or EMPTY).shared or EMPTY
+        end,
+      }
+      local loader = lazy_loaders[key]
+      if not loader then
+        if lua_enabled and not sandbox_enabled then
+          return _G[key]
+        end
+        return
+      end
+      -- set the result on the table to not load again
+      local value = loader()
+      rawset(self, key, value)
+      return value
+    end,
+    __newindex = function(self)
+      error("This environment is read-only.")
+    end,
+  }
+
+  local template_env = {}
+  if lua_enabled and sandbox_enabled then
+    -- load the sandbox environment to be used to render the template
+    template_env = deep_copy(sandbox.configuration.environment)
+    -- here we can optionally add functions to expose to the sandbox, eg:
+    -- tostring = tostring,
+    -- because headers may contain array elements such as duplicated headers
+    -- type is a useful function in these cases. See issue #25.
+    template_env.type = type
+  end
+  setmetatable(template_env, __meta_environment)
+  local compiled_resource_attributes = {}
+  for current_name, current_value in pairs(resource_attributes) do
+    local res, err = param_value(current_value, resource_attributes, template_env)
+    if err then
+      ngx_log(ngx_DEBUG, _log_prefix, "failed to render the template " ..
+            current_value .. ", error:" .. err)
+      return nil
```

sorry about my previous suggestion being incomplete, since this is a utils function, let's just propagate the error, we can then log an `ngx.ERR` level log from the caller, if there is an error.
```suggestion
      return nil, err
```